htdocs/fourn/class/fournisseur.facture.class.php

```diff
@@ -1848,6 +1848,11 @@ public function validate($user, $force_number = '', $idwarehouse = 0, $notrigger
 							$result = $mouvP->reception($user, $this->lines[$i]->fk_product, $idwarehouse, $this->lines[$i]->qty, $up_ht_disc, $langs->trans("InvoiceValidatedInDolibarr", $num));
 						}
 						if ($result < 0) {
+							if (count($mouvP->errors)) {
+								setEventMessages(null, $mouvP->errors, 'errors');
```

Comment:
We are inside a method than can be called out of user interface context, for example by a batch.
So we should not have setEventMessage inside such method.
Instead, you must set error message inside
this->error
And return a negative value
Then the caller canndecide to use setEvent... Or doing something else depending on user context.

Reply:
Hello @eldy ,

We have taken your comments into consideration.

Can you check this **PR**, and see if it can be merged?

Thanks